src/golang/cmd/server/handler/get_operator_result.go

```diff
@@ -93,21 +95,33 @@ func (h *GetOperatorResultHandler) Perform(ctx context.Context, interfaceArgs in
 
 	emptyResp := shared.ExecutionState{}
 
-	dbOperatorResult, err := h.OperatorResultReader.GetOperatorResultByWorkflowDagResultIdAndOperatorId(
+	dbWorkflowDagResult, err := h.WorkflowDagResultReader.GetWorkflowDagResult(
 		ctx,
 		args.workflowDagResultId,
-		args.operatorId,
 		h.Database,
 	)
 	if err != nil {
-		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error occurred when retrieving operator result.")
+		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error occurred when retrieving workflow result.")
 	}
 
-	response := shared.ExecutionState{
-		Status: dbOperatorResult.Status,
+	response := shared.ExecutionState{}
+	dbOperatorResult, err := h.OperatorResultReader.GetOperatorResultByWorkflowDagResultIdAndOperatorId(
+		ctx,
+		args.workflowDagResultId,
+		args.operatorId,
+		h.Database,
+	)
+	if err != nil {
+		if err != database.ErrNoRows {
+			return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error occurred when retrieving operator result.")
+		}
+		// OperatorResult was never created, so we use the WorkflowDagResult's status as this OperatorResult's status
+		response.Status = dbWorkflowDagResult.Status
```

Comment:
should the entire status be a `failure` with some logs like `the result is not created`? I think this depends on when these results get created during dag execution